src/uri-template.js

```diff
@@ -16,7 +16,13 @@ export default function buildUriTemplate(basePath, href, pathObjectParameters =
     const parameterNames = _.unique([].concat(pathObjectParameterNames, queryParameterNames));
     const parameterNamesString = parameterNames.length ? `{?${parameterNames.join(',')}}` : '';
 
-    return `${basePath}${href}${parameterNamesString}`;
+    const full = `${basePath}${href}${parameterNamesString}`;
+
+    // Before returning, we replace instances of `-` with `%2d`, but only when
+    // they occur inside of a template variable.
+    return full.replace(/\{.*?\}/g, (match) => {
```

Comment:
I've updated the test to show that this works as expected. It only replaces things within `{` and `}` (non-greedy), which should only get variables. We can't just do the query vars because the path itself may contain variables, e.g. `pets/{id}{?foo}`.